### DISCUSSION TITLE
No power - nothing - need some help

### DISCUSSION THREAD

**Discussion by PFalko** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#545] [Q&A]
Hi,

i have a Miami S100101 from pre 2021 with 6 wires, i guess. 
I order the PCB and uploaded everything and can (with usb power) access the webpage. I also put the rest of the hardware in place, but i am unsure if this is ok. Why? Well, if i put everything together nothing happens. The board doesnt even have power.

![2](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/63845442/0d065bd0-35a2-47c3-9b83-585e25f8a7a2)
![1](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/63845442/3ec3271d-1d05-4a43-aaf3-286722ea7095)

Attached is the original setup and what i but in between. I already tryed switching the in and out cables DSP/CIO (bc from the pictures male/female connection was confusing me).

Any ideas?

#1 Yes, i need to use the proper JST SM instead of PH.
#2 HWTest says:

CIO to DSP pin 3 success!
CIO to DSP pin 4 success!
CIO to DSP pin 5 success!

DSP to CIO pin 3 success!
DSP to CIO pin 4 success!
DSP to CIO pin 5 success!

Should i see the other pins here too?

Regards,
Falko

**Comment by visualapproach** [comment]
Hi @PFalko it seems you have put the cables in the wrong port. The top slot (H1A+B) on the PCB where it says CIO should go to the lower part of the pump. The other slot (H2A+B) that says DSP should go to the display. Also make sure you put them in the right orientation. Measure with a voltmeter that you have 5v between "5V" and "GND" as marked on the A-ports.
The hardware test is fine. Your solder joints are working.
Cheers! I had enough coffee today anyway ;-)

**Reply by PFalko** [reply]
Allright, i fixed this - thank you! I was able to get some positive results after adjusting the hardware version to the 2021 model and fixing the wires. I also managed to get the buttons working and some simple timers.

However, there are still some issues/questions:

#1 Wireless access. I dont have any network available outside of the civilization in the wilderness.
-> For my point of view it the device can stay in AP mode, but needs to function (assuming i setup the time and date manual). 
Is this possible right now/in the future? From what i read i need an external wifi till the device finished booting and for every reboot. Right?
What else can i do to keep it working? Setup and empty wireless access point with no internet? I could use a raspi for this maybe.

#2 The device switches "always" back to "°F" instead of staying at "°C". Using "restore last values" doesnt help (i assume it would).
Is it possible to bootstrap some setting, which will be guaranteed after every reboot? Something like Autostart values?
Is this possible right now? If not i would like to raise a feature request, e.g. implementing an menu item for this and execute the commands after boot and reboot.

#3 Can i somehow export/import scheduled commands? || Can i edit scheduled commands?
Right now i can only wipe all commands, but not edit/delete a certain one. Maybe i miss some information (e.g. edit those via the build in file browser?)? If thats not possible i would like to raise a feature request.

#4 Settings backup
Is it possible to export (and import later) all settings to a file?

#5 Is it possible to implement routines?
For example i have in mind the following:

I. pump and heat 
input: start time + duration + repeat interval + desired temperature
desired outcome: power on device, do pump and heat as told and after this power off device
I know i can manual achieve this, but see #3 - if there is an error i have to start over. :(

II. pump
input: start time + duration + repeat interval
desired outcome: power on device, do pump as told and after this power off device
I know i can manual achieve this, but see #3 - if there is an error i have to start over. :(

**Reply by visualapproach** [reply]
> Allright, i fixed this - thank you! I was able to get some positive results after adjusting the hardware version to the 2021 model and fixing the wires. I also managed to get the buttons working and some simple timers.
> 
> 
> 
> However, there are still some issues/questions:
> 
> 
> 
> #1 Wireless access. I dont have any network available outside of the civilization in the wilderness.
> 
> -> For my point of view it the device can stay in AP mode, but needs to function (assuming i setup the time and date manual). 
> 
> Is this possible right now/in the future? From what i read i need an external wifi till the device finished booting and for every reboot. Right?
> 
> What else can i do to keep it working? Setup and empty wireless access point with no internet? I could use a raspi for this maybe.

After you have setup everything you can turn off AP in the web ui. It will not start the autoportal if no Wi-Fi present and it should work. 


> #2 The device switches "always" back to "°F" instead of staying at "°C". Using "restore last values" doesnt help (i assume it would).
> 
> Is it possible to bootstrap some setting, which will be guaranteed after every reboot? Something like Autostart values?
> 
> Is this possible right now? If not i would like to raise a feature request, e.g. implementing an menu item for this and execute the commands after boot and reboot.

It should remember the unit setting. May not work outside the pump (disconnected from pump)
If it persists, file a bug report

> #3 Can i somehow export/import scheduled commands? || Can i edit scheduled commands?

Not yet


> Right now i can only wipe all commands, but not edit/delete a certain one. Maybe i miss some information (e.g. edit those via the build in file browser?)? If thats not possible i would like to raise a feature request.
> 
> 
> 
> #4 Settings backup
> 
> Is it possible to export (and import later) all settings to a file?

You can copy them from /dir/ but maybe you need to hard reset after upload to avoid them being overwritten

> #5 Is it possible to implement routines?
> 
> For example i have in mind the following:
> 
> 
> 
> I. pump and heat 
> 
> input: start time + duration + repeat interval + desired temperature
> 
> desired outcome: power on device, do pump and heat as told and after this power off device
> 
> I know i can manual achieve this, but see #3 - if there is an error i have to start over. :(
> 
> 
> 
> II. pump
> 
> input: start time + duration + repeat interval
> 
> desired outcome: power on device, do pump as told and after this power off device
> 
> I know i can manual achieve this, but see #3 - if there is an error i have to start over. :(


I'll probably make the command queue editable. However I'm rebuilding my sun deck now and spare time is not in stock.

**Reply by PFalko** [reply]
#1 
It will auto reconnect wifi if its available again or is a reboot required? Can i press a keysequence to trigger a wifi reconnect?
#2
Ok, but what about "autostart"?

**Reply by visualapproach** [reply]
When starting, the ESP looks for the known Wi-Fi and connects if it finds it. If not it will start the autoportal access point IF that option is checked in the config. So if you uncheck Autoportal it will periodically scan for the known Wi-Fi, but work locally. It will not find an NTP so command queue time may be off by a great amount or will not work (I can't remember). Restoring last states after a power outage should work though, but you better test before relying on it.
Restore states after start should reset the unit to last setting. File a bug report if you want.
